src/components/checkboxFilter/index.tsx

```diff
@@ -18,34 +18,55 @@ function CheckboxFilter<ItemKey extends string, FilterName extends string>({
   const hasGroups = items.some(
     (item) => (item.childCheckboxes ?? []).length > 0,
   );
+
+  const itemsPerColumn = Math.ceil(items.length / numberOfColumnsOnDesktop);
+
+  // Create an array to store items grouped by columns
+  const columns = Array.from(
+    { length: numberOfColumnsOnDesktop },
+    (_, columnIndex) => {
+      const columnStartIndex = columnIndex * itemsPerColumn;
+      const columnEndIndex = Math.min(
+        (columnIndex + 1) * itemsPerColumn,
+        items.length,
+      );
+      return [...items].slice(columnStartIndex, columnEndIndex);
+    },
+  );
+
   return (
     <TranslationProvider language={language} scopes={['checkboxFilter']}>
       <Box
         sx={{
-          columns: { md: numberOfColumnsOnDesktop, base: 1 },
-          columnRule: 'solid var(--chakra-colors-gray-100) 1px',
-          columnGap: 'var(--chakra-space-4xl)',
+          display: 'grid',
+          gridTemplateColumns: {
+            base: '1fr',
+            md: `repeat(${numberOfColumnsOnDesktop}, 1fr)`,
+          },
+          gap: 'var(--chakra-space-4xl)',
         }}
       >
-        {items.map((item) => {
-          if (item.childCheckboxes && item.childCheckboxes.length > 0)
-            return (
-              <CheckboxGroupCollapsibleWithChildren
-                key={item.key}
-                item={item}
-                onApply={onApply}
-                onToggleCheckboxGroup={onToggleCheckboxGroup}
-              />
-            );
-          return (
-            <CheckboxWithFacet
-              key={item.key}
-              item={item}
-              onApply={onApply}
-              indentFacet={hasGroups}
-            />
-          );
-        })}
+        {columns.map((columnItems, columnIndex) => (
+          <div key={columnIndex} data-testid="column">
+            {columnItems.map((item) => (
+              <div key={item.key} data-testid={`item-${item.key}`}>
```

Comment:
would it also work without the additional div or has it a purpose besides having a test-id?

Reply:
I think it would work, but I wanted a test to be there for easier queries and to be covered.